plugins/logs/plugin.go

```diff
@@ -780,3 +795,26 @@ func (p *Plugin) logEvent(event EventV1) error {
 	}).Info("Decision Log")
 	return nil
 }
+
+func deepCopy(obj interface{}) (interface{}, error) {
+	switch v := obj.(type) {
+	case map[string]interface{}:
+		var buf bytes.Buffer
+		enc := gob.NewEncoder(&buf)
+		dec := gob.NewDecoder(&buf)
+		if err := enc.Encode(v); err != nil {
+			return nil, err
+		}
+		var objCopy map[string]interface{}
+		if err := dec.Decode(&objCopy); err != nil {
+			return nil, err
+		}
+		return interface{}(objCopy), nil
+	default:
+		return nil, fmt.Errorf("unexpected type when decoding during deep copy")
+	}
+}
+
+func init() {
+	gob.Register(map[string]interface{}{})
+}
```

Comment:
There's a manual deep copy implementation in storage/inmem/txn.go that will work for this case where the result types are known: https://github.com/open-policy-agent/opa/blob/master/storage/inmem/txn.go#L391. Let's use that one instead. It handles the case where the result is a slice and should be quite a bit faster. If you don't mind, take the implementation from txn.go and move it into an internal package like `internal/deepcopy` and then make use of it here.